app/services/exports/case_log_export_service.rb

```diff
@@ -20,57 +31,123 @@ def is_omitted_field?(field_name)
       field_name.starts_with?("details_known_") || pattern_age.match(field_name) || omitted_attrs.include?(field_name) ? true : false
     end
 
-    LOG_ID_OFFSET = 300_000_000_000
-
   private
 
-    def save_export_run(current_time)
+    def get_daily_run_number
       today = Time.zone.today
-      last_daily_run_number = LogsExport.where(created_at: today.beginning_of_day..today.end_of_day).maximum(:daily_run_number)
-      last_daily_run_number = 0 if last_daily_run_number.nil?
+      LogsExport.where(created_at: today.beginning_of_day..today.end_of_day).count + 1
+    end
+
+    def build_export_run(current_time, full_update)
+      if LogsExport.count.zero?
+        return LogsExport.new(started_at: current_time)
+      end
+
+      base_number = LogsExport.maximum(:base_number)
+      increment_number = LogsExport.where(base_number:).maximum(:increment_number)
 
-      export = LogsExport.new
-      export.daily_run_number = last_daily_run_number + 1
-      export.started_at = current_time
-      export
+      if full_update
+        base_number += 1
+        increment_number = 1
+      else
+        increment_number += 1
+      end
+
+      LogsExport.new(started_at: current_time, base_number:, increment_number:)
     end
 
-    def write_master_manifest(export)
+    def write_master_manifest(daily_run, archive_datetimes)
       today = Time.zone.today
-      increment_number = export.daily_run_number.to_s.rjust(4, "0")
+      increment_number = daily_run.to_s.rjust(4, "0")
       month = today.month.to_s.rjust(2, "0")
       day = today.day.to_s.rjust(2, "0")
       file_path = "Manifest_#{today.year}_#{month}_#{day}_#{increment_number}.csv"
-      string_io = build_manifest_csv_io
+      string_io = build_manifest_csv_io(archive_datetimes)
       @storage_service.write_file(file_path, string_io)
     end
 
-    def write_export_data(case_logs)
-      string_io = build_export_xml_io(case_logs)
-      file_path = "#{get_folder_name}/#{get_file_name}.xml"
-      @storage_service.write_file(file_path, string_io)
+    def get_archive_name(case_log, base_number, increment)
+      collection_start = case_log.collection_start_year
+      month = case_log.startdate.month
```

Comment:
We might have to skip any log where we do not have a startdate (collection year will be nil too)